app/controllers/api/v1/orders_controller.rb

```diff
@@ -130,6 +130,7 @@ def order_record
         if order_params[:stockit_id]
           @order = Order.accessible_by(current_ability).where(stockit_id: order_params[:stockit_id]).first_or_initialize
           @order.assign_attributes(order_params)
+          @order.assign_state(params[:status])
```

Comment:
@abulsayyad123  does this get run for Stockit requests or also for normal goodcity requests?

Reply:
Only for Stockit requests.